run_service.sh

```diff
@@ -665,17 +665,27 @@ if [ "$rcp_new_filter_supported" = false ]; then
 fi
 
 echo "RPC checks passed."
+
+echo ""
+echo "------------------------------"
+echo "Setting up '$directory' repository"
+echo "------------------------------"
 echo ""
 
-# clone repo
-if [ -d $directory ]
-then
-    echo "Detected an existing $directory directory. Using this one..."
-    echo "Please stop and manually delete the $directory repo if you updated the service's version ($service_version)!"
-    echo "You can run the following command, or continue with the pre-existing version of the service:"
-    echo "rm -r $directory"
-else
-    echo "Cloning the $directory repo from $org_name GitHub..."
+if [ -d "$directory" ]; then
+    current_version=$(git --git-dir="$directory/.git" describe --tags)
+
+    if [ "$current_version" != "$service_version" ]; then
+        echo "Current version of $directory ($current_version) does not match expected version ($service_version)."
+        echo "Removing '$directory' directory..."
+        echo ""
+        sudo rm -rf "$directory"
+    fi
```

Comment:
Maybe prompt the user to confirm that they do not mind losing any changes in the code.

Reply:
I am unsure if this is required. @dagacha please confirm.

Reply:
IMO it is not required. As an operator I need to actively download the new version in the repository for the code to change. So when I run the script I do it under the assumption that I want to change the code, so no further confirmation needed.

Reply:
Confirmed to work on Ubuntu. This is whay you see:

`------------------------------
Setting up 'trader' repository
------------------------------

Current version of trader (v0.11.4) does not match expected version (v0.11.5).
Removing 'trader' directory...
`

Reply:
Confirmed to work on MacOs 14.2